pkg/openapi/validation.go

```diff
@@ -239,8 +240,45 @@ func (o *Controller) getGVKByDefinitionName(definitionName string) (gvk string,
 	return "", preferredGVK, fmt.Errorf("gvk not found by the given definition name %s, %v", definitionName, versionsTyped.gvks)
 }
 
+func parseGVK(str string) (group, apiVersion, kind string) {
+	if strings.Count(str, "/") == 0 {
+		return "", "", str
+	}
+	splitString := strings.Split(str, "/")
+	if strings.Count(str, "/") == 1 {
+		return "", splitString[0], splitString[1]
+	}
+	return splitString[0], splitString[1], splitString[2]
+}
+
+func groupMatches(gvkMap map[string]bool, group, kind string) bool {
+	if group == "" {
+		ok := gvkMap["core"]
+		if ok {
+			return true
+		}
+	} else {
+		elements := strings.Split(group, ".")
+		for _, item := range elements {
+			if item != "io" {
+				ok := gvkMap[item]
+				if ok {
+					return true
```

Comment:
Should we only check the first `item` in `elements` and return immediately? The logic is cleaner this way.

For example, we are checking if the first element of group `iam` exists in the definitionName.
```
definitionName: "io.crossplane.gcp.iam.v1alpha1.ServiceAccount",		
gvk: "iam.gcp.crossplane.io/v1alpha1/ServiceAccount",
```

Reply:
okay